pkg/prometheus/promcfg.go

```diff
@@ -92,11 +92,28 @@ func NewConfigGenerator(logger log.Logger, p monitoringv1.PrometheusInterface, e
 		return nil, fmt.Errorf("failed to parse scrape classes: %w", err)
 	}
 
+	endpointSliceSupportedAndConfigured := false // Always assume false to preserve original prometheus-operator behaviour.
+
+	// Check if the user has explicitly set the service discovery role to use.
+	switch serviceDiscoveryRole := *cpf.ServiceDiscoveryRole; serviceDiscoveryRole {
```

Comment:
```suggestion
	switch serviceDiscoveryRole := ptr.Deref(cpf.ServiceDiscoveryRole, monitoringv1.Endpoints); serviceDiscoveryRole {
```

Reply:
Thank you for your suggestion. I have implemented this in [7c1adf2](https://github.com/prometheus-operator/prometheus-operator/pull/6672/commits/7c1adf2433cbba2fb6c1bd84d736450767c03110)